data_pipeline/dataset.py

```diff
@@ -150,68 +161,93 @@ def __len__(self):
     return len(self.ids)
 
   def __getitem__(self, item):
-    """Returns: id, sentence, concepts, adj_mat, amr_str."""
+    """Returns: id, sentence, sentence_str, concepts, concepts_str, adj_mat, amr_str."""
     id = self.ids[item]
     sentence = self.fields_by_id[id][SENTENCE_KEY]
+    sentence_str = self.fields_by_id[id][SENTENCE_STR_KEY]
     concepts = self.fields_by_id[id][CONCEPTS_KEY]
+    concepts_str = self.fields_by_id[id][CONCEPTS_STR_KEY]
     glove_concepts = self.fields_by_id[id][GLOVE_CONCEPTS_KEY]
     adj_mat = self.fields_by_id[id][ADJ_MAT_KEY]
     amr_str = self.fields_by_id[id][AMR_STR_KEY]
-    return id, sentence, concepts, glove_concepts, adj_mat, amr_str
+    return id, sentence, sentence_str, concepts, concepts_str, glove_concepts, adj_mat, amr_str
 
   def collate_fn(self, batch):
     amr_ids = []
     batch_sentences = []
+    batch_sentences_strings = []
     batch_concepts = []
+    batch_concepts_strings = []
     batch_glove_concepts = []
     batch_adj_mats = []
     amr_strings = []
     sentence_lengths = []
     concepts_lengths = []
     for entry in batch:
-      amr_id, sentence, concepts, glove_concepts, adj_mat, amr_str = entry
+      amr_id, sentence, sentence_str, concepts, concepts_str, glove_concepts, adj_mat, amr_str = entry
       amr_ids.append(amr_id)
       batch_sentences.append(sentence)
+      batch_sentences_strings.append(sentence_str)
       batch_concepts.append(concepts)
+      batch_concepts_strings.append(concepts_str)
       batch_glove_concepts.append(glove_concepts)
       batch_adj_mats.append(adj_mat)
       amr_strings.append(amr_str)
       sentence_lengths.append(len(sentence))
       concepts_lengths.append(len(concepts))
     # Get max lengths for padding.
     max_sen_len = max([len(s) for s in batch_sentences])
+    max_sen_str_len = max([len(s) for s in batch_sentences_strings])
     max_concepts_len = max([len(s) for s in batch_concepts])
+    max_concepts_str_len = max([len(s) for s in batch_concepts_strings])
     max_glove_concepts_len = max([len(s) for s in batch_glove_concepts])
     max_adj_mat_size = max([len(s) for s in batch_adj_mats])
     # Pad sentences.
     padded_sentences = [
       torch_pad(s, (0, max_sen_len - len(s))) for s in batch_sentences]
+    # Pad initial sentences.
+    padded_initial_sentences = []
+    for sentence in batch_sentences_strings:
+        padded_sentence = copy.deepcopy(sentence)
+        for i in range(max_sen_str_len - len(sentence)):
+            padded_sentence.append(PAD)
+        padded_initial_sentences.append(padded_sentence)
+
```

Comment:
could be moved into a documented method
~ the same code follows shortly for concepts